azext_iot/central/params.py

```diff
@@ -386,3 +384,75 @@ def load_central_arguments(self, _):
             options_list=["--description", "--desc"],
             help="Detailed description of the job.",
         )
+
+    with self.argument_context("iot central query") as context:
+        context.argument(
+            "query_string",
+            options_list=["--query-string", "--qs"],
+            help="Query clause to retrieve telemetry or property data.",
+        )
+        context.argument(
+            "api_version",
+            options_list=["--api-version", "--av"],
+            choices=CaseInsensitiveList([ApiVersion.v1_1_preview.value]),
+            default=ApiVersion.v1_1_preview.value,
+            help="The API version for the requested operation.",
+        )
+
+    with self.argument_context("iot central export destination") as context:
+        context.argument(
+            "destination_id",
+            options_list=["--dest-id"],
+            help="Unique identifier for the export destination.",
+        )
+
+    with self.argument_context("iot central export destination create") as context:
+        context.argument(
+            "content",
+            options_list=["--content", "-k"],
+            help="The destination definition. Provide path to JSON file or raw stringified JSON."
+            " [File Path Example:./path/to/file.json]"
+            " [Example of stringified JSON:{<Destination Data JSON>}]. The request body must contain content of Destination.",
+        )
+
+    with self.argument_context("iot central export destination update") as context:
+        context.argument(
+            "content",
+            options_list=["--content", "-k"],
+            help="The partial destination definition. Provide path to JSON file or raw stringified JSON."
+            " [File Path Example:./path/to/file.json]"
+            " [Example of stringified JSON:{<Destination Data JSON>}]."
+            " The request body must contain partial content of Destination.",
+        )
+
+    with self.argument_context("iot central export") as context:
+        context.argument(
+            "export_id",
+            options_list=["--export-id"],
+            help="Unique identifier for the export.",
+        )
+        context.argument(
+            "api_version",
+            options_list=["--api-version", "--av"],
+            choices=CaseInsensitiveList([ApiVersion.v1_1_preview.value]),
+            default=ApiVersion.v1_1_preview.value,
+            help="The API version for the requested operation.",
+        )
+
+    with self.argument_context("iot central export create") as context:
+        context.argument(
+            "content",
+            options_list=["--content", "-k"],
```

Comment:
I know these commands are preview, so we can potentially change the design in a future iteration.

We should consider a design where minimal JSON input is required where possible. The common properties for an export definition or export destination definition can be added as CLI parameters, and the CLI can assemble the json payload. This makes the commands easier to use improving devEx.

The CLI can also do work in finding connection strings or other elements for integration -- for example Digital Twins  endpoint creation has operations per endpoint type https://docs.microsoft.com/en-us/cli/azure/dt/endpoint/create?view=azure-cli-latest#az_dt_endpoint_create_eventhub

Reply:
I highly second this. I gave some ideas on how to break up content in some other comments with help. If the arguments get messy, you can also use arg groups to help bucket the arguments in a reasonable manner.

Reply:
separated the payload for both destination & export create command.